src/components/reset-password.tsx

```diff
@@ -0,0 +1,159 @@
+                        <Group justify="space-between" gap="xl">
+                            <Text ta="left">Reset Your Password</Text>
+                            <CloseButton aria-label="Close form" onClick={() => router.push('/')} />
+                        </Group>
+                    </Paper>
+                    <Paper bg="#f5f5f5" shadow="none" p={30} radius="sm">
+                        <Text>Enter the verification code from your email and choose a new password.</Text>
+                        <TextInput
+                            key={verificationForm.key('code')}
+                            {...verificationForm.getInputProps('code')}
+                            label="Verification Code"
+                            placeholder="Enter code from email"
+                            aria-label="Verification code"
+                        />
+                        <PasswordInput
+                            key={verificationForm.key('password')}
+                            {...verificationForm.getInputProps('password')}
+                            label="New Password"
+                            placeholder="Enter new password"
+                            aria-label="New password"
+                            mt={10}
+                        />
+                        <Stack align="center" mt={15}>
+                            <Button type="submit">Reset Password</Button>
+                            <Anchor onClick={() => setPendingReset(null)}>Back to Email Entry</Anchor>
+                        </Stack>
+                    </Paper>
+                </form>
+            </Stack>
+        )
+    }
+
+    return (
+        <Stack>
+            <form onSubmit={emailForm.onSubmit((values) => onSubmitEmail(values))}>
+                <Paper bg="#d3d3d3" shadow="none" p={10} mt={30} radius="sm">
+                    <Group justify="space-between" gap="xl">
+                        <Text ta="left">Reset Password</Text>
+                        <CloseButton aria-label="Close form" onClick={() => router.push('/')} />
+                    </Group>
+                </Paper>
+                <Paper bg="#f5f5f5" shadow="none" p={30} radius="sm">
+                    <Text>Enter your email address and we&#39;ll send you a verification code.</Text>
+                    <TextInput
+                        key={emailForm.key('email')}
+                        {...emailForm.getInputProps('email')}
+                        label="Email"
+                        placeholder="Email address"
+                        aria-label="Email address"
+                    />
+                    <Stack align="center" mt={15}>
+                        <Button type="submit">Send Reset Code</Button>
+                        <Anchor onClick={() => router.push('/')}>Back to Login</Anchor>
+                    </Stack>
+                </Paper>
+            </form>
+        </Stack>
+    )
+}
```

Comment:
it feels to me like this component is really two separate components,.

Maybe we could split most of it into other components and then this could be:

```ts
export function ResetPassword() {
     const { isLoaded, signIn, setActive } = useSignIn()
     const [pendingReset, setPendingReset] = useState<Boolean>(false)

     if (pendingReset) return <PendingReset />

     return <ResetForm onComplete={() => setPendingReset(true)} />
}
```